packages/NodeTypeResolver/PHPStan/Scope/NodeVisitor/ByRefVariableNodeVisitor.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\NodeTypeResolver\PHPStan\Scope\NodeVisitor;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\AssignRef;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\FunctionLike;
+use PhpParser\NodeVisitorAbstract;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\PhpDocParser\NodeTraverser\SimpleCallableNodeTraverser;
+
+final class ByRefVariableNodeVisitor extends NodeVisitorAbstract
+{
+    public function __construct(
+        private readonly SimpleCallableNodeTraverser $simpleCallableNodeTraverser
+    ) {
+    }
+
+    public function enterNode(Node $node): ?Node
+    {
+        if (! $node instanceof FunctionLike) {
+            return null;
+        }
+
+        $stmts = $node->getStmts();
+        if ($stmts === null) {
+            return null;
+        }
+
+        $byRefVariableNames = [];
+        if ($node instanceof Closure) {
+            foreach ($node->uses as $closureUse) {
+                if ($closureUse->byRef && is_string($closureUse->var)) {
+                    $closureUse->var->setAttribute(AttributeKey::IS_BYREF_VAR, true);
```

Comment:
why do we need a attribute on the close when the closure already has a byRef property?

Reply:
that to bring byref from closure use:

```php
$byRef = 'old value';

function () use (&$byRef) {
     $byRef = 'some value';
}

echo $byRef;
```

That's different from param:

```php
$byRef = 'old value';

function foo(&$byRef) {
   $byRef = 'some value';
}
```